### ISSUE TITLE
Can't get focus out of text box on web.whatsapp.com

### ISSUE THREAD

**Issue by bovine3dom** [tridactyl/tridactyl#3070]
Escape and `<C-,>` don't successfully unfocus the text box (but the commands do run).

Further investigation:

```
document.activeElement.blur() // unfocuses the text box if run in the console
setTimeout(x=>document.activeElement.blur(),3000) // if you move the mouse back onto the page, focus doesn't leave the box
setTimeout(x=>document.activeElement.blur(),3000) // if you keep the mouse on the firefox console, focus does leave the box
```

:man_shrugging: 

Perhaps related to #1567.

**Comment by hp77-creator** [comment]
There are problems on other chat service based websites like, I will post what I experienced on my machine:
OS: Ubuntu 20.04
Firefox: 83.0(64 bit)

In messenger facebook using on firefox:
Seems like, the issues are not as uniform in other chat services(like web.telegram or messenger) as they are in web.whatsapp. There seems to be some issue in detecting which pane or tab inside a window a user is focusing, for example, If I am scrolling up in a particular chat and the chat is not loaded then it won't scroll up any further but the focus of the cursor(or the appropriate term) shifts to the list of chats and scrolls them up. This seems like not that big of an issue or is it? I don't know, Tell me I will do the necessary. I was writing a comment to the issue when I thought I should I give it a fresh try with <C-,> The focus gets out of the text box but it shifts to the list of chats rather than in the particular chat. What will be better or maybe a feature, if we can navigate between the list and the particular chat. There exist hint mode to access a particular chat but navigation is not distinct between list of chats and particular chat pane or tab.

**Comment by bovine3dom** [comment]
Thanks for posting this here.

As I said on Matrix, 

> scrolling up changing frame when you have reached the top is on purpose, we didn't think about infinite upwards scrolling when we made it
> we could probably make a setting that disables it and you could just disable it for certain sites

**Comment by m2habert** [comment]
I came up with a temporary workaround just the other day using `hint -f`... It disengages from the input mode when used with an unmatched string. I then chose to follow with the cmd-line:

       `bindurl web.whatsapp.com  --mode=input     <C-CR> composite hint -f m2 ; fillcmdline `
       `bindurl web.whatsapp.com  --mode=insert    <C-CR> composite hint -f m2 ; fillcmdline`

**Comment by m2habert** [comment]
update:   even better workaround : 

            bindurl web.whatsapp.com  --mode=normal         <Escape> composite hint -f m2 ; fillcmdline ; ex.hide_and_clear
            bindurl web.whatsapp.com  --mode=insert         <Escape> composite hint -f m2 ; fillcmdline ; ex.hide_and_clear
            bindurl web.whatsapp.com  --mode=input          <Escape> composite hint -f m2 ; fillcmdline ; ex.hide_and_clear

... where `hint -f` is the "fixed input" version of `hint` and almost any random input will return a non matched hint (in my case, I use "m2"), which will in turn incur in disengagement from the hint mode and returning to the normal mode.

**Comment by ntc2** [comment]
This is still a problem, and the #1567 linked in the description was closed a few years ago, so I guess that didn't solve it.

**Comment by bovine3dom** [comment]
I don't think that issue was about focus - it was about hints not working. Does ctrl+, not work for focus?

**Comment by ntc2** [comment]
No, `ctrl+,` does not work to remove focus from the message compose area on web.whatsapp.com. The [workaround above](https://github.com/tridactyl/tridactyl/issues/3070#issuecomment-1080227866) works well, altho sometimes I need to hit `Esc` twice (sometimes the first time I press `Esc` it enters some kind of hint following mode, but with very limited choice of hints, and then the second `Esc` returns to normal mode with nothing focused).
